expo/gbe/views/review_bid_view.py

```diff
@@ -0,0 +1,95 @@
+from django.views.generic import View
+from django.utils.decorators import method_decorator
+from django.contrib.auth.decorators import login_required
+from django.shortcuts import (
+    render,
+    get_object_or_404,
+)
+from django.http import HttpResponseRedirect
+from django.core.urlresolvers import reverse
+from expo.gbe_logging import log_func
+from gbe.forms import (
+    BidEvaluationForm,
+    BidStateChangeForm,
+    PersonaForm,
+    ClassBidForm,
+    ParticipantForm,
+
+)
+from gbe.models import (
+    Class,
+    BidEvaluation,
+)
+from gbe.functions import (
+    validate_perms,
+    get_conf,
+)
+
+
+class ReviewBidView(View):
+    bid_state_change_form = BidStateChangeForm
+
+    def create_action_form(self, bid):
+        self.actionform = self.bid_state_change_form(instance=bid)
+        self.actionURL = reverse(self.changestate_view_name,
+                                 urlconf='gbe.urls',
+                                 args=[bid.id])
+
+
+    def bid_review_response(self, request):
+        return render(request,
+                      'gbe/bid_review.tmpl',
+                      {'readonlyform': self.readonlyform_pieces,
+                       'reviewer': self.reviewer,
+                       'form': self.form,
+                       'actionform': self.actionform,
+                       'actionURL': self.actionURL,
+                       'conference': self.conference,
+                       'old_bid': self.old_bid,
+                       })
+
+    def create_object_form(self, initial={}):
+        self.object_form = self.bid_form_type(instance=self.object,
+                                  prefix=self.bid_prefix,
+                                  initial=initial)
+
+    def post_response_for_form(self, request):
+        if self.form.is_valid():
+            evaluation = self.form.save(commit=False)
+            evaluation.evaluator = self.reviewer
+            evaluation.bid = self.object
+            evaluation.save()
+            return HttpResponseRedirect(reverse(self.review_list_view_name,
+                                                urlconf='gbe.urls'))
+        else:
+            return self.bid_review_response(request)
+
+    def object_not_current_redirect(self):
+        if self.object.is_current:
+            return None
+        return HttpResponseRedirect(
+            reverse(self.bid_view_name, urlconf='gbe.urls', args=[self.object.id]))
+
+    def get_object(self, request, object_id):
+        self.object = get_object_or_404(self.object_type,
+                                     id=object_id)
+
+    def groundwork(self, request, args, kwargs):
```

Comment:
I'm wishing that there was a way to define:
- self.readonlyform_pieces 
- self.form

on the parent class level.  It's very much hidden that these element MUST be there for bid_review_response to be successful.  Anyone adding a new bid class would need to know this.

I'd like to either - developer-proof the parent class so that if the defaults aren't overridden a best-effort workable result occurs.

-- OR-- 
provide painfully obvious documentation at the top of this class documenting what the must-have functions and data members are when you are subclassing and what they must do.